src/main/java/org/folio/edge/core/EdgeVerticle2.java

```diff
@@ -73,7 +75,15 @@ public void start(Future<Void> future) {
 
     secureStore = initializeSecureStore(config().getString(SYS_SECURE_STORE_PROP_FILE));
 
-    final HttpServer server = getVertx().createHttpServer();
+    // enable response compression if property 'response_compression' = true,
+    // disable compression if 'response_compression' = false or not exists at all
+    final HttpServerOptions serverOptions = new HttpServerOptions();
+    if (config().containsKey(SYS_RESPONSE_COMPRESSION)) {
+      serverOptions.setCompressionSupported(config().getBoolean(SYS_RESPONSE_COMPRESSION));
```

Comment:
Hi @mreno-EBSCO ,

thank you very much for your useful advice.

I think the most effective way now is to follow your idea to set other properties via system properties while default deployment options are predefined in Constants.java. Regarding to response compression this means it should be disabled by default in Constants.java (we can enable it by “response_compression” = true in the deployment properties). In case it is enabled further compression strategy will be determined by request headers.

There are two possible cases here.
- Request doesn’t contain header Encoding-Type/contains only Encoding-Type: instance;
In this case response will be returned without compression (this covered by two unit tests).
- Request contains header with Encoding-Type: instance, gzip, deflate.

Response is returned with compression type choosing by Vert.x algorithm (this covered by separate tests for gzip and deflate).

It seems we don’t need to have additional possibility to change response compression on-the-fly during unit tests. Instead that I prepared new unit tests class where compression verification provided.

Please, review my PR again and let me know yours objections if needed.

Thanks,
Slava